src/VisualStudio/Core/Def/DocumentOutline/DocumentSymbolDataViewModel.cs

```diff
@@ -0,0 +1,150 @@
+    {
+        public string Name { get; }
+        public ImmutableArray<DocumentSymbolDataViewModel> Children { get; }
+        public int StartPosition => RangeSpan.Start;
+        public SnapshotSpan RangeSpan { get; }
+        public SnapshotSpan SelectionRangeSpan { get; }
+        public SymbolKind SymbolKind { get; }
+        public ImageMoniker ImageMoniker { get; }
+
+        private bool _isExpanded;
+        public bool IsExpanded
+        {
+            get => _isExpanded;
+            set => SetProperty(ref _isExpanded, value);
+        }
+
+        private bool _isSelected;
+        public bool IsSelected
+        {
+            get => _isSelected;
+            set => SetProperty(ref _isSelected, value);
+        }
+
+        public DocumentSymbolDataViewModel(DocumentSymbolData documentSymbolData, ImmutableArray<DocumentSymbolDataViewModel> children)
+            : this(
+                  documentSymbolData.Name,
+                  children,
+                  documentSymbolData.RangeSpan,
+                  documentSymbolData.SelectionRangeSpan,
+                  documentSymbolData.SymbolKind,
+                  GetImageMoniker(documentSymbolData.SymbolKind),
+                  isExpanded: true,
+                  isSelected: false)
+        {
+        }
+
+        private DocumentSymbolDataViewModel(
+            string name,
+            ImmutableArray<DocumentSymbolDataViewModel> children,
+            SnapshotSpan rangeSpan,
+            SnapshotSpan selectionRangeSpan,
+            SymbolKind symbolKind,
+            ImageMoniker imageMoniker,
+            bool isExpanded,
+            bool isSelected)
+        {
+            Name = name;
+            Children = children;
+            SymbolKind = symbolKind;
+            ImageMoniker = imageMoniker;
+            _isExpanded = isExpanded;
+            _isSelected = isSelected;
+            RangeSpan = rangeSpan;
+            SelectionRangeSpan = selectionRangeSpan;
+        }
+
+        private void NotifyPropertyChanged([CallerMemberName] string? propertyName = null)
+            => PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(propertyName));
+
+        public event PropertyChangedEventHandler? PropertyChanged;
+
+        private void SetProperty(ref bool field, bool value, [CallerMemberName] string propertyName = "")
+        {
+            // Note: we do not lock here. Worst case is that we fire multiple
+            //       NotifyPropertyChanged events which WPF can handle.
+            if (field == value)
+            {
+                return;
+            }
+
+            field = value;
+            NotifyPropertyChanged(propertyName);
+        }
+
+        internal DocumentSymbolDataViewModel WithChildren(ImmutableArray<DocumentSymbolDataViewModel> newChildren)
+            => new(Name, newChildren, RangeSpan, SelectionRangeSpan, SymbolKind, ImageMoniker, IsExpanded, IsSelected);
```

Comment:
trying to figure out when the children change... but the range of it wouldn't change...

Reply:
This is only used in sorting where we change the order of the children but nothing else.